routes/users.js

```diff
@@ -0,0 +1,83 @@
+const express = require("express");
+
+const { Package, Order } = require("../model/Package");
+const { User } = require("../model/User");
+const { verifyJWTToken } = require("../middlewares/verifyJWTtoken");
+
+const { isDuplicate } = require("../utils/isDuplicate");
+
+const router = express.Router();
+
+router.post("/:userId/bookmark", async (req, res, next) => {
+  try {
+    const userId = req.params.userId;
+    const { action } = req.body;
+
+    const existUser = await User.findById(userId).lean();
+    if (!existUser) {
+      return res.status(404).json({ error: "존재하지 않는 사용자입니다." });
+    }
+
+    const userBookmark_idList = existUser.bookmark;
+
+    const userWithBookmarks = await User.findById(userId)
+      .populate("bookmark")
+      .lean();
+    const bookmarkList = userWithBookmarks.bookmark;
+
+    const isNewBookmark = !bookmarkList.some((bookmark) =>
+      isDuplicate(bookmark, action),
+    );
+    if (!userBookmark_idList.length || isNewBookmark) {
+      const newBookmark = await Order.create(action);
+
+      await User.updateOne(
+        { _id: userId },
+        { $push: { bookmark: newBookmark } },
+      );
```

Comment:
updateOne의 대해서도 반환값을 받아서 update 됐는지 안됐는에 대한 처리를 할 수 있습니다~!😁
```suggestion
      const result = await User.updateOne(
        { _id: userId },
        { $push: { bookmark: newBookmark } },
      );
      
      if (!result.acknowledged) {
        // 업데이트 실패했을 경우~
      }
```

Reply:
updateOne에 이런 반환값이 있었군요...! 활용해보도록 하겠습니다! 좋은 의견 감사합니다!